### PR TITLE
refactor: RuntimeType creation and member resolution

### PR DESCRIPTION
**Pull request by Velfi**
[Apologies for the large PR](https://www.youtube.com/watch?v=ax27VgI-lVE)

## Motivation and Context

Maintainability and ease of use

## Description

The short version:

- refactor: `RuntimeType` creation and member resolution
- refactor: `RuntimeType` companion functions
- add: `AwsCargoDependency` package
- refactor: move aws runtime crate funs into `AwsRuntimeType`
- update: all code affected by the above changes

The long version:
This PR implements the another part of #1933.

- `runtimeCrate` has been renamed to `smithyRuntimeCrate` and now requires users to pass the "smithy-" prefix as part of the crate name:
  before: `val smithyHttp = runtimeCrate("http").asType()`
  after: `val smithyHttp = smithyRuntimeCrate("smithy-http").asType()`
- `awsRuntimeDependency` has been renamed to `awsRuntimeCrate` in order to match `smithyRuntimeCrate`.
- Functions returning `RuntimeType`s have all been consolidated into static methods and fields in the `RuntimeType` companion object . AWS-specific `RuntimeType` functions have been centralized in `software.amazon.smithy.rustsdk.AwsRuntimeType`.
- If you want to access a Cargo dependency, use one of the companion methods/fields on `CargoDependency` and `AwsCargoDependency`. The `awsRuntimeCrate`/`smithyRuntimeCrate` functions still exist but you should avoid using them.

## Testing

Ran existing tests

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by ysaito1001 — APPROVED** [review]
Thanks for making extensive changes. I like how the new codebase is more consistent & readable.

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt [hunk 1]
```diff
@@ -174,7 +174,18 @@ fun Project.registerGenerateCargoConfigTomlTask(
...
+            } catch (ex: Exception) {
+                println("failed to write to '.cargo/config.toml' with exception $ex")
+            }
+
+            if (outputDir.resolve(".cargo/config.toml").createNewFile()) {
+                println("created '.cargo/config.toml'")
+            } else {
+                println("failed to create '.cargo/config.toml'")
+            }
+
```
[the remark this answers is not in the document]
Comment (Velfi) [reply buildSrc/src/main/kotlin/CodegenTestCommon.kt]: I removed this change b/c the server team has committed to fix the issue.

The issue is that when running `./gradlew :codegen-server-test:python:cargoTest`
the command fails for me with:
```
java.io.FileNotFoundException: /Users/zhessler/Documents/smithy-rs/codegen-server-test/python/build/smithyprojections/codegen-server-test-python/.cargo/config.toml (No such file or directory)
```

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt [hunk 2]
```diff
@@ -174,7 +174,18 @@ fun Project.registerGenerateCargoConfigTomlTask(
     this.tasks.register("generateCargoConfigToml") {
         description = "generate `.cargo/config.toml`"
         doFirst {
-            outputDir.resolve(".cargo").mkdir()
+            try {
+                outputDir.resolve(".cargo").mkdir()
+            } catch (ex: Exception) {
+                println("failed to write to '.cargo/config.toml' with exception $ex")
```
[the remark this answers is not in the document]
Comment (Velfi) [reply buildSrc/src/main/kotlin/CodegenTestCommon.kt]: I removed this change b/c the server team has committed to fix the issue.

File: buildSrc/src/main/kotlin/CodegenTestCommon.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Velfi) [reply buildSrc/src/main/kotlin/CodegenTestCommon.kt]: Trying it now, it works. I'm assuming they fixed it somehow. That or it's just a spurious failure and I was particularly unlucky yesterday.

**Review by crisidev — APPROVED** [review]
LGTM.

**Pull request merged**